codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenUtils.java

```diff
@@ -326,4 +329,28 @@ public static CollectionShape expectCollectionShape(Shape shape) {
 
         throw new CodegenException("expect shape " + shape.getId() + " to be Collection, was " + shape.getType());
     }
+
+    public static final class SortedMembers implements Comparator<MemberShape> {
+        private final Model model;
+        private final SymbolProvider symbolProvider;
+
+        public SortedMembers(Model model, SymbolProvider symbolProvider) {
+            this.model = model;
+            this.symbolProvider = symbolProvider;
+        }
+
+        @Override
+        public int compare(MemberShape a, MemberShape b) {
+            int ai = a.getMemberTrait(model, RequiredTrait.class).isPresent() ? 1 : 0;
```

Comment:
You don't need to use `getMemberTrait` here since only members can be marked with the Required trait. This also means you can remove the need to have a `Model` to create this class.

Reply:
Thanks looks like `hasTrait` is the correct method to use instead, right?

Reply:
Yeah